### PR TITLE
Added Dedicated Game Server Sample (experimental)

### PR DESCRIPTION
**Pull request by RikuTheFuffs**
### Description
This PR includes a new experimental sample that showcases how to use official Unity packages and tools to make a multiplayer game.

### Issue Number(s)
N/A

### Contribution checklist
 - [N/A] Tests have been added for the project and/or any internal package
 - [N/A] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [N/A] JIRA ticket ID is in the PR title or at least one commit message
 - [N/A] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=160) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=160) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: fernando-cortez<br/>:white_check_mark: RikuTheFuffs<br/>:white_check_mark: jilfranco-unity<br/>:x: Louis-Philippe Lafontaine-Bédard<br/><hr/>**Louis-Philippe Lafontaine-Bédard** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=160) it.</sub>

File: .yamato/_triggers.yml
[the remark this answers is not in the document]
Comment (fernando-cortez) [reply .yamato/_triggers.yml:1]: Yeah, disregard the comments about project-standard.yml as that only is part of develop atm!

> Is this something specific to the DGS sample and that was introduced in this PR?

It seems like it is. _triggers.yml and tests.yml are introduced here, and add project tests, whereas before status checks before only had the projects being built.

> Or is this a general improvement that could be done in a separate PR?

For the purposes of getting this PR through, I'll create another PR to reduce the number of jobs here. I truly think we don't need a build job on top of a project test job (the latter should depend on the former's artifacts).

> I also see that recently it was added a Run all tasks on the bitesize sample develop branch (head) when there is a push to the Netcode for GameObjects develop branch which is totally unneeded and puts a lot of strain on our CI.

This is fine to me btw, we've been discussing how we need these triggers to notify if our projects break per Netcode for GameObjects's develop branch's changes. This is something @fluong6 will look into to, so we can leave this one for now as well.

File: .yamato/_triggers.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+# Run all relevant tasks when a pull request targets the develop / main or a sample branch
+pull_request_trigger:
+  name: Pull Request Trigger (main, develop, & sample branches)
+  dependencies:
+{% for project in projects -%}
+{% for editor in project.test_editors -%}
+{% for platform in test_platforms -%}
+    - .yamato/build.yml#build_{{ project.name }}_{{ editor }}_{{ platform.name }}
```
Comment (fernando-cortez) [inline .yamato/_triggers.yml:11]: This is the line that I feel is redundant. I'll investigate in a future PR, after integrating into develop.

**Review by fernando-cortez — COMMENTED** [review]
We'll have to reinvestigate the Yamato jobs created, but for a future PR. Seems good otherwise though.

Have @jilfranco-unity's comments from #159 been addressed here? If so, I'll approve right away!

Edit: I also edited the Yamato Pull Request status check on the repo settings for PRs against main since the name changed inside this PR. Just a heads-up.

**Comment by RikuTheFuffs** [comment]
> Have @jilfranco-unity's comments from #159 been addressed here? If so, I'll approve right away!

@fernando-cortez  yes!

> On that note too, CLA is still a blocker. You can temporarily disable that before closing, reopen, and re-enable to workaround this issue.

I have the permissions to bypass branch protections, it should be enough to merge

File: .yamato/_triggers.yml
[the remark this answers is not in the document]
Comment (RikuTheFuffs) [reply .yamato/_triggers.yml:1]: > I'll create another PR to reduce the number of jobs here. I truly think we don't need a build job on top of a project test job (the latter should depend on the former's artifacts).

Sounds good, and I agree. Not sure why they were added in the first place.

> This is fine to me btw, we've been discussing how we need these triggers to notify if our projects break per Netcode for GameObjects's develop branch's changes. This is something @fluong6 will look into to, so we can leave this one for now as well.

About this, wouldn't it make more sense to test against their `Main` branch, so we know if a new release breaks the sample?
After all, `develop ` is, by definition, an unstable branch where things change quickly.

[the remark this answers is not in the document]
Comment (fernando-cortez) [reply .yamato/_triggers.yml:1]: > About this, wouldn't it make more sense to test against their Main branch, so we know if a new release breaks the sample?
> After all, develop  is, by definition, an unstable branch where things change quickly.

I hear you. Let's discuss that with QA and get their opinion on it. My gut is that if we have resources to run these checks, why not.

**Review by fernando-cortez — APPROVED** [review]
Looks great! Happy to have this in! 🎉

**Pull request merged**